modules/_labs/select/react/lib/SelectMenu.tsx

```diff
@@ -1,18 +1,47 @@
-import * as React from 'react';
+import React, {useState, useEffect} from 'react';
+import {Rect} from '@popperjs/core';
+
 import {CSSObject, keyframes} from '@emotion/core';
 import {EmotionCanvasTheme, ErrorType, Themeable, styled} from '@workday/canvas-kit-react-common';
-import {colors, borderRadius, inputColors, spacingNumbers} from '@workday/canvas-kit-react-core';
+import {Placement, Popper, useCloseOnEscape} from '@workday/canvas-kit-react-popup';
+import {colors, borderRadius, inputColors} from '@workday/canvas-kit-react-core';
+
 import {SelectProps} from './Select';
+import {buttonBorderWidth} from './SelectBase';
 
 interface SelectMenuProps
   extends Themeable,
     React.HTMLAttributes<HTMLUListElement>,
     Pick<SelectProps, 'error'> {
+  /**
+   * The ref to the anchor button element. Required to anchor the menu.
+   */
+  buttonRef: React.RefObject<HTMLButtonElement>;
   /**
    * If true, enable animation on the SelectMenu.
    * @default true
    */
   isAnimated: boolean;
+  /**
+   * If true, automatically flip the SelectMenu to keep it visible if necessary (e.g., if the the SelectMenu would otherwise display below the visible area of the viewport).
+   * @default true
+   */
+  isAutoFlipped: boolean;
+  /**
+   * If true, focus the SelectMenu when it's shown. Set to false if you don't want to focus the SelectMenu automatically (for visual testing purposes, for example).
+   * @default true
+   */
+  isAutoFocused: boolean;
+  /**
+   * If true, flip the SelectMenu so it extends upwards from the button.
+   * @default false
+   */
+  isFlipped: boolean;
```

Comment:
I think I'd prefer to have a disjoint union of the options allowed:

```ts
placement: 'top-start' | 'bottom-start'
```

Or since `top` and `bottom` are the only ones with meaning since we make the width the same as the parent:

```ts
placement: 'top' | 'bottom'
```

Reply:
What would be the advantage of a union over a boolean in this case?

Reply:
Booleans convey less meaning and context that string literals. What does "isFlipped" do? Is positioning on the top of the target mean `isFlipped=true` or `isFlipped=false`? Sure, the JSDoc may explain what `true` and `false` mean, but now you have to go to the docs every time. It just is less obvious and thus more mental energy to understand.

Reply:
+1. I was also confused by the terminology "flipped"